src/modules/qt/gps_parser.cpp

```diff
@@ -686,71 +773,68 @@ void xml_parse_gpx(xmlNodeSetPtr found_nodes, gps_point_ll **gps_list, int *coun
 		</extensions>
 	</trkpt>
 	*/
-
-	//mlt_log_info(NULL, "xml_parse_gpx - parsing %d elements\n", found_nodes->nodeNr);
-
-	for (i=0; i<found_nodes->nodeNr; i++)
+	while (!reader.atEnd() && !reader.hasError())
 	{
-		gps_point_raw crt_point = uninit_gps_raw_point;
-		crt_node = found_nodes->nodeTab[i];
-
-		if (xmlHasProp(crt_node, _x("lat"))) {
-			xmlChar* str = xmlGetProp(crt_node, _x("lat"));
-			crt_point.lat = strtod(_s(str), NULL);
-			xmlFree(str);
-		}
-		if (xmlHasProp(crt_node, _x("lon"))) {
-			xmlChar* str = xmlGetProp(crt_node, _x("lon"));
-			crt_point.lon = strtod(_s(str), NULL);
-			xmlFree(str);
-		}
-		for (gps_val = crt_node->children; gps_val; gps_val = gps_val->next)
+		reader.readNext();
+		// mlt_log_info(NULL, " readNext tag: %s type = %d\n", _qxml_getthestring(reader.name()), reader.tokenType());
+		if (reader.isStartElement() && reader.name() == "trkpt")
 		{
-			if (strncmp(_s(gps_val->name), "ele", strlen("ele")) == 0)
-				crt_point.ele = strtod(_s(gps_val->children->content), NULL);
-			else if (strncmp(_s(gps_val->name), "time", strlen("time")) == 0)
-				crt_point.time = datetimeXMLstring_to_mseconds(_s(gps_val->children->content), NULL);
-			else if (strncmp(_s(gps_val->name), "bearing", strlen("bearing")) == 0)
-				crt_point.bearing = (int)strtod(_s(gps_val->children->content), NULL);
-			else if (strncmp(_s(gps_val->name), "speed", strlen("speed")) == 0)
-				crt_point.speed = strtod(_s(gps_val->children->content), NULL);
-			else if (strncmp(_s(gps_val->name), "extensions", strlen("extensions")) == 0)
+			gps_point_raw crt_point = uninit_gps_raw_point;
+			
+			QXmlStreamAttributes attributes = reader.attributes();
+			if (attributes.hasAttribute("lat"))
+				crt_point.lat = attributes.value("lat").toDouble();
+			if (attributes.hasAttribute("lon"))
+				crt_point.lon = attributes.value("lon").toDouble();
+			
+			while (reader.readNext() && !(reader.name() == "trkpt" && reader.tokenType() == QXmlStreamReader::EndElement)) //until closing trkpt
```

Comment:
compile error on Qt 6:
`../src/modules/qt/gps_parser.cpp:790:51: error: conversion from ‘const char [6]’ to ‘QChar’ is ambiguous`